crates/runtime/src/instance.rs

```diff
@@ -211,7 +211,7 @@ impl Instance {
 
     /// Return the indexed `VMMemoryDefinition`.
     fn memory_ptr(&self, index: DefinedMemoryIndex) -> *mut VMMemoryDefinition {
-        unsafe { self.vmctx_plus_offset(self.offsets.vmctx_vmmemory_definition(index)) }
+        unsafe { *self.vmctx_plus_offset(self.offsets.vmctx_vmmemory_pointer(index)) }
```

Comment:
This I think has pretty deep implications which require an audit of callers of this function to ensure they still work correctly. Looking I see:

* `Instance::memory` - this is not ok for shared memories since it does a non-atomic read of the pointer
* `Instance::set_memory` - this is no longer valid and needs to be updated. Callers of this function are:
  * `Instance::memory_grow` - this'll need to be updated since the update only happens for owned memories
* `Instance::get_exported_memory` - this is ok but I think callers of this need to be updated to do atomic reads of the length field
* `Instance::memory_index` - this is no longer valid and needs to be updated (or deleted)